core/services/ocrcommon/adapters.go

```diff
@@ -71,3 +80,110 @@ func (c *OCR3ContractTransmitterAdapter) Transmit(ctx context.Context, digest oc
 func (c *OCR3ContractTransmitterAdapter) FromAccount() (ocrtypes.Account, error) {
 	return c.ct.FromAccount()
 }
+
+var _ ocr3types.OnchainKeyring[[]byte] = (*OCR3OnchainKeyringMultiChainAdapter)(nil)
+
+type OnchainSigningStrategy interface {
```

Comment:
@vyzaldysanchez Can we avoid defining this as an interface? I'm worried about the coupling this introduces on lines 98-109: you're passing in the strategy and then immediately calling two methods on it.

Can we just pass in a map[string]ocr2key.KeyBundle directly, having extracted the relevant config beforehand? This will result in a bit more boiler plate in the original callsite, but a cleaner implementation overall.